facebook/ThreatExchange
# Terraform Shared Cognito User Pool

Summary
---------

Added terraforming of a shared Cognito user pool and the ability to use it in the main terraform flow. Also updated some documentation.

Test Plan
---------

1. In a terminal go to `/terraform/authentication-standalone`
2. Set up terraform.tfvars (set organization to fb)
3. Run `terraform apply`
4. Use the outputs (`webapp_and_api_shared_user_pool_id` and `webapp_and_api_shared_user_pool_client_id`) in `/terraform/terraform.tfvars` and set `use_shared_user_pool` to `true` (see `/terraform/terraform.tfvars.example`).
5. Run `terraform apply` from `/terraform`
6. Create a user in the shared user pool
7. Go to the webapp s3 bucket and open the static website url from the Properties tab
8. Test signing in to the webapp
9. Regression testing (test with `use_shared_user_pool` = `false`, both with and without a cloudfront distribution)
